pkg/reconciler/nscert/nscert.go

```diff
@@ -0,0 +1,204 @@
+		c.Logger.Errorf("Failed to apply domain template %s to domain %s and namespace %s: %v", cfg.Network.DomainTemplate, defaultDomain, ns.Name, err)
+		return err
+	}
+
+	desiredCert := resources.MakeWildcardCertificate(ns, dnsName, defaultDomain)
+	labelSelector := kubelabels.SelectorFromSet(
+		kubelabels.Set{
+			networking.WildcardCertDomainLabelKey: defaultDomain,
+		},
+	)
+
+	// If any labeled cert has been issued for our DNSName then there's nothing to do
+	existingCerts, err := c.knCertificateLister.Certificates(ns.Name).List(labelSelector)
+	if err != nil {
+		return fmt.Errorf("Failed to list certificates: %v", err)
+	}
+
+	matchingCert := findMatchingCert(dnsName, existingCerts)
+	if matchingCert != nil {
+		return nil
+	}
+
+	// If there is no matching cert find one previously created by this reconciler which may
+	// need to be updated.
+	existingCert, err := findNamespaceCert(ns, existingCerts)
+
+	if apierrs.IsNotFound(err) {
+		cert, err := c.ServingClientSet.NetworkingV1alpha1().Certificates(ns.Name).Create(desiredCert)
+		if err != nil {
+			c.Logger.Errorw("Failed to create namespace certificate", zap.Error(err))
+			c.Recorder.Eventf(ns, corev1.EventTypeWarning, "CreationFailed",
+				"Failed to create Knative certificate %s/%s: %v", ns.Name, desiredCert.ObjectMeta.Name, err)
+			return err
+		}
+
+		c.Recorder.Eventf(cert, corev1.EventTypeNormal, "Created",
+			"Created Knative Certificate %s/%s", ns.Name, cert.ObjectMeta.Name)
+	} else if err != nil {
+		return err
+	} else if !metav1.IsControlledBy(existingCert, ns) {
+		existingCert.Status.MarkResourceNotOwned("CertManagerCertificate", existingCert.Name)
+		return fmt.Errorf("Namespace %s does not own Knative Certificate: %s", ns.Name, existingCert.Name)
+	} else if !equality.Semantic.DeepEqual(existingCert.Spec, desiredCert.Spec) {
+		copy := existingCert.DeepCopy()
+		copy.Spec = desiredCert.Spec
+		_, err := c.ServingClientSet.NetworkingV1alpha1().Certificates(copy.Namespace).Update(copy)
+		if err != nil {
+			c.Logger.Errorw("Failed to update Knative Certificate", zap.Error(err))
+			c.Recorder.Eventf(existingCert, corev1.EventTypeWarning, "UpdateFailed",
+				"Failed to update Knative Certificate %s/%s: %v", desiredCert.Namespace, desiredCert.Name, err)
+			return err
+		}
+		c.Recorder.Eventf(existingCert, corev1.EventTypeNormal, "Updated",
+			"Updated Spec for Knative Certificate %s/%s", desiredCert.Namespace, desiredCert.Name)
+		return nil
+	}
+
+	return nil
+}
+
+func wildcardDomain(tmpl, domain, namespace string) (string, error) {
+	data := network.DomainTemplateValues{
+		Name:      "*",
+		Domain:    domain,
+		Namespace: namespace,
+	}
+
+	buf := bytes.Buffer{}
+	t, err := template.New("domain-template").Parse(tmpl)
+	if err != nil {
+		return "", err
+	}
+	if err := t.Execute(&buf, data); err != nil {
+		return "", err
+	}
+
+	dom := buf.String()
+	if !domainTemplateRegex.Match([]byte(dom)) {
+		return "", errors.New("Invalid DomainTemplate")
```

Comment:
more debug information, like value of the template? At least log it.

Reply:
I've added the value of dom to the error message